### PR TITLE
final mask nc file creation

### PR DESCRIPTION
**Pull request by Velip09**

### CODE REVIEW

File: src/pyadps/utils/writenc.py [hunk 1]
```diff
@@ -307,6 +309,9 @@ def finalnc(outfile, depth, time, data, t0="hours since 2000-01-01", attributes=
     evel.units = "cm/s"
     evel.long_name = "error_velocity"
 
+    mvel = ncfile.createVariable("Final_mask", np.float32, ("time", "depth"), fill_value=fill)
```
Comment (p-amol) [inline src/pyadps/utils/writenc.py:312]: Change NetCDF variable name to "mask".

Comment (p-amol) [reply src/pyadps/utils/writenc.py:312]: See if you can add the following as a long name for the `mask`: 
``
mvel.long_name = "Velocity Mask (1: bad value, 0: good value)"
``
Else replace `mvel.long_name` with `mvel.comment`.

File: src/pyadps/utils/writenc.py [hunk 2]
```diff
@@ -315,6 +320,7 @@ def finalnc(outfile, depth, time, data, t0="hours since 2000-01-01", attributes=
     vvel[:, :] = data[1, :, :].T
     wvel[:, :] = data[2, :, :].T
     evel[:, :] = data[3, :, :].T
+    mvel[:, :] = final_mask.T
     
     # Add global attributes if provided
```
Comment (p-amol) [inline src/pyadps/utils/writenc.py:325]: Add a Global attribute to indicate whether or not the mask is applied to the data. 

``mask_applied = "True"``

**Review by p-amol — CHANGES_REQUESTED** [review]

File: src/pyadps/utils/writenc.py (hunk 2 above)
Comment (Velip09) [reply src/pyadps/utils/writenc.py:325]: Sir, where this mask_applied we come in which file?

File: src/pyadps/utils/writenc.py (hunk 1 above)
Comment (Velip09) [reply src/pyadps/utils/writenc.py:312]: okay

File: src/pyadps/utils/writenc.py (hunk 2 above)
Comment (p-amol) [reply src/pyadps/utils/writenc.py:325]: `ncfile.mask_applied = "True"`

Add it to the end.

Comment (p-amol) [reply src/pyadps/utils/writenc.py:325]: I have merged Aravind's branch, which has resolved the mask issue. The conflict should be resolved before merging.

**Pull request closed without merging**